### PR TITLE
New tool to list and count in place families

### PR DESCRIPTION
**Pull request by jmcouffin**
that is a first for me, so bare with me and let me know if there is something wrong with the code or the pull request

### CODE REVIEW

**Comment by eirannejad** [comment]
Merged into PR branch to review and test.

**Comment by eirannejad** [comment]
Made a few modifications and merged into develop. Will be pushed with the next release. See https://github.com/eirannejad/pyRevit/commit/b6c5bfdef28bdce4c83d01796933d427cac19d93

Thanks a bunch for a great tool

**Comment by jmcouffin** [comment]
Cool. Thanks.
I will learn looking at your adjustments.

Le mar. 12 nov. 2019 19:17, Ehsan Iran-Nejad <notifications@github.com> a
écrit :

File: extensions/pyRevitTools.extension/pyRevit.tab/Analysis.panel/Tools.stack/Analyse.pulldown/Get In Place Family Count.pushbutton/script.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+Get a total count and types of In Place Family in the current document.
+Copyright (c) 2019 Jean-Marc Couffin
+https://github.com/jmcouffin
+--------------------------------------------------------
+PyRevit Notice:
+Copyright (c) 2014-2019 Ehsan Iran-Nejad
+pyRevit: repository at https://github.com/eirannejad/pyRevit
+"""
+
+__title__ = 'Get In Place Family count'
```
Comment (eirannejad) [inline extensions/pyRevitTools.extension/pyRevit.tab/Analysis.panel/Tools.stack/Analyse.pulldown/Get In Place Family Count.pushbutton/script.py:11]: If the bundle name is identical to `__title__` you don't need to set this. pyRevit uses the bundle name as default.

File: extensions/pyRevitTools.extension/pyRevit.tab/Analysis.panel/Tools.stack/Analyse.pulldown/Get In Place Family Count.pushbutton/script.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+__author__ = 'Jean-Marc Couffin'
+__contact__ = 'https://github.com/jmcouffin'
+__credits__ = 'http://eirannejad.github.io/pyRevit/credits/'
+__doc__ = 'Get a total count and types of In Place Family in the current document.'
```
Comment (eirannejad) [inline extensions/pyRevitTools.extension/pyRevit.tab/Analysis.panel/Tools.stack/Analyse.pulldown/Get In Place Family Count.pushbutton/script.py:15]: pyRevit grabs the docstring from the top of the script file it exists. No need to duplicate and redefine `__doc__`

File: extensions/pyRevitTools.extension/pyRevit.tab/Analysis.panel/Tools.stack/Analyse.pulldown/Get In Place Family Count.pushbutton/script.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+--------------------------------------------------------
+PyRevit Notice:
+Copyright (c) 2014-2019 Ehsan Iran-Nejad
+pyRevit: repository at https://github.com/eirannejad/pyRevit
+"""
+
+__title__ = 'Get In Place Family count'
+__author__ = 'Jean-Marc Couffin'
+__contact__ = 'https://github.com/jmcouffin'
+__credits__ = 'http://eirannejad.github.io/pyRevit/credits/'
```
Comment (eirannejad) [inline extensions/pyRevitTools.extension/pyRevit.tab/Analysis.panel/Tools.stack/Analyse.pulldown/Get In Place Family Count.pushbutton/script.py:14]: Not necessary. It's already covered in the About tool

File: extensions/pyRevitTools.extension/pyRevit.tab/Analysis.panel/Tools.stack/Analyse.pulldown/Get In Place Family Count.pushbutton/script.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+__doc__ = 'Get a total count and types of In Place Family in the current document.'
+
+
+from pyrevit.framework import List
+from pyrevit import revit, DB
+
+famInstE = DB.FilteredElementCollector(revit.doc)\
```
Comment (eirannejad) [inline extensions/pyRevitTools.extension/pyRevit.tab/Analysis.panel/Tools.stack/Analyse.pulldown/Get In Place Family Count.pushbutton/script.py:21]: use [snake case](https://en.wikipedia.org/wiki/Snake_case) convention in python e.g. `family_instance_collector`

File: extensions/pyRevitTools.extension/pyRevit.tab/Analysis.panel/Tools.stack/Analyse.pulldown/Get In Place Family Count.pushbutton/script.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+             .OfClass(DB.FamilyInstance)\
+             .WhereElementIsNotElementType()\
+             .ToElements()
+             
+famInstInPlace = [e for e in famInstE if e.Symbol.Family.IsInPlace]
```
Comment (eirannejad) [inline extensions/pyRevitTools.extension/pyRevit.tab/Analysis.panel/Tools.stack/Analyse.pulldown/Get In Place Family Count.pushbutton/script.py:26]: Use `x` in comprehensions. It's a best practice.

File: extensions/pyRevitTools.extension/pyRevit.tab/Analysis.panel/Tools.stack/Analyse.pulldown/Get In Place Family Count.pushbutton/script.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+print('In Place Families count : {0}'.format(len(list(famInstInPlace))))
```
Comment (eirannejad) [inline extensions/pyRevitTools.extension/pyRevit.tab/Analysis.panel/Tools.stack/Analyse.pulldown/Get In Place Family Count.pushbutton/script.py:28]: `famInstInPlace` is already a list (created with list comprehension) not need to `list(famInstInPlace)` again

File: extensions/pyRevitTools.extension/pyRevit.tab/Analysis.panel/Tools.stack/Analyse.pulldown/Get In Place Family Count.pushbutton/script.py [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+elist = []
+
+print('\nCATEGORY & FAMILY NAME')
+
+for e in famInstInPlace:
+    ptcategory = revit.doc.GetElement(revit.doc.GetElement(e.Id).GetTypeId()).Category.Name
```
Comment (eirannejad) [inline extensions/pyRevitTools.extension/pyRevit.tab/Analysis.panel/Tools.stack/Analyse.pulldown/Get In Place Family Count.pushbutton/script.py:35]: Try to break down long lines like this and verify the inputs so your tool doesn't break. e.g. Get the type using `revit.doc.GetElement(e.Id).GetTypeId()` first and verify there is one before getting the Category

File: extensions/pyRevitTools.extension/pyRevit.tab/Analysis.panel/Tools.stack/Analyse.pulldown/Get In Place Family Count.pushbutton/script.py [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    ename = e.Symbol.Family.Name
+    l = ptcategory +' | '+ ename
+    elist.append(l)
```
Comment (eirannejad) [inline extensions/pyRevitTools.extension/pyRevit.tab/Analysis.panel/Tools.stack/Analyse.pulldown/Get In Place Family Count.pushbutton/script.py:38]: You can print as you are looping through families.

**Comment by eirannejad** [comment]
@jmcouffin Left you a few minor comments. :D

**Comment by jmcouffin** [comment]
Thank you so much for taking the time to do so @eirannejad.  That helps immensely to understand do's and don't and improve. Very encouraging!

File: extensions/pyRevitTools.extension/pyRevit.tab/Analysis.panel/Tools.stack/Analyse.pulldown/Get In Place Family Count.pushbutton/script.py (hunk 5 above)
Comment (jmcouffin) [reply extensions/pyRevitTools.extension/pyRevit.tab/Analysis.panel/Tools.stack/Analyse.pulldown/Get In Place Family Count.pushbutton/script.py:26]: in here, 

`inplace_families = [
    x for x in familyinstance_collector
    if x.Symbol and x.Symbol.Family and x.Symbol.Family.IsInPlace
    ]` 

why do you split the if statement in 3 blocks? probably a noob question.

File: extensions/pyRevitTools.extension/pyRevit.tab/Analysis.panel/Tools.stack/Analyse.pulldown/Get In Place Family Count.pushbutton/script.py (hunk 8 above)
Comment (jmcouffin) [reply extensions/pyRevitTools.extension/pyRevit.tab/Analysis.panel/Tools.stack/Analyse.pulldown/Get In Place Family Count.pushbutton/script.py:38]: I thought about having the link made through the id, did not know it was implemented as nicely. neat feature.

File: extensions/pyRevitTools.extension/pyRevit.tab/Analysis.panel/Tools.stack/Analyse.pulldown/Get In Place Family Count.pushbutton/script.py (hunk 5 above)
Comment (eirannejad) [reply extensions/pyRevitTools.extension/pyRevit.tab/Analysis.panel/Tools.stack/Analyse.pulldown/Get In Place Family Count.pushbutton/script.py:26]: To make sure `x.Symbol` is not `None` before pulling the `Family` property and `x.Symbol.Family` is not `None` before pulling `IsInPlace`. Always verify your data

**Comment by jmcouffin** [comment]
That makes sense especially when I  think about all the list management in
my dynamo graphs to deal with nulls or none or empty values.  Thanks

Le jeu. 14 nov. 2019 19:39, Ehsan Iran-Nejad <notifications@github.com> a
écrit :

**Pull request merged**